lib/time.ml

```diff
@@ -84,16 +89,16 @@ let in_the_past now t =
   t = 0L || t <= now ()
 
 let rec restart_threads now =
-  match SleepQueue.lookup_min !sleep_queue with
-    | Some{ canceled = true; _ } ->
-        sleep_queue := SleepQueue.remove_min !sleep_queue;
-        restart_threads now
-    | Some{ time = time; thread = thread; _ } when in_the_past now time ->
-        sleep_queue := SleepQueue.remove_min !sleep_queue;
-        Lwt.wakeup thread ();
-        restart_threads now
-    | _ ->
-        ()
+  match SleepQueue.maximum sleep_queue with
+  | { canceled = true; _ } ->
+      SleepQueue.remove sleep_queue;
+      restart_threads now
+  | { time = time; thread = thread; _ } when in_the_past now time ->
+      SleepQueue.remove sleep_queue;
+      Lwt.wakeup thread ();
+      restart_threads now
+  | _ -> ()
+  | exception Not_found -> ()
```

Comment:
I'd put the `exception` case at the beginning, and as reported by @mato, this should be `exception Binary_heap.empty` instead of `Not_found` (same below).